nautobot/utilities/testing/views.py

```diff
@@ -747,6 +749,30 @@ def test_list_objects_with_permission(self):
                 response = self.client.get(f"{self._get_url('list')}?export")
                 self.assertHttpStatus(response, 200)
                 self.assertEqual(response.get("Content-Type"), "text/csv")
+                instance1 = self._get_queryset().first()
+                # With filtering
+                response = self.client.get(f"{self._get_url('list')}?export&id={instance1.pk}")
+                self.assertHttpStatus(response, 200)
+                self.assertEqual(response.get("Content-Type"), "text/csv")
+                response_body = response.content.decode(response.charset)
+                # I cannot count the number of lines "\n" directly because sometimes there might be too many characters
+                # in a line that will cause a linewrap, so we might end up with more than 2 lines
+                count = 0
+                csv_file = StringIO(response_body)
+                reader = csv.reader(csv_file, delimiter=",")
+                for _ in reader:
+                    count += 1
+                self.assertEqual(count, 2)
+                # This block of code is for checking the content of the csv output.
+                # But it does not work for some of the models.
+                # csv_headers = getattr(self.model, "csv_headers")
+                # model_field = (
+                #     "name" or "slug" or "username"
+                #     if ("name" or "username" or "slug") in csv_headers
+                #     else csv_headers[0]
+                # )
+                # self.assertIn(str(getattr(instance1, model_field)), response_body)
+                # self.assertIn(str(getattr(instance2, model_field)), response_body)
```

Comment:
I need some help figuring out a way to check the content of the csv response based on `instance1` attributes for all models.

Reply:
```python
response_body = response.content.decode(response.charset)
reader = csv.DictReader(StringIO(response_body))
data = [dict(row) for row in reader]


# Sample response
# [{"name": "Device 1", ...}, ...]
```
I believe with this format you can check what ever you need to check

Reply:
Note: `instance1_csv_data` do not account for custom fields, which might be found in `data`